presto-main/src/main/java/com/facebook/presto/operator/aggregation/state/BigIntegerAndLongState.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.aggregation.state;
+
+import java.math.BigInteger;
+
+@AccumulatorStateMetadata(stateFactoryClass = BigIntegerAndLongStateFactory.class, stateSerializerClass = BigIntegerAndLongStateSerializer.class)
+public interface BigIntegerAndLongState
+        extends AccumulatorState
+{
+    BigInteger getBigInteger();
```

Comment:
no, in this form this `AccumulatorState` implementation is just a holder for a `BigInteger` and `long`, as you see in it's name, it doesn't know what those values represent

I was thinking about it, there are two kinds of `AccumulatorState` classes in Presto. Some are domain named, like `CorrelationState` and some are like `LongState`, `LongAndDoubleState`, I chose to follow the latter, but I'm not strongly convinced about it.